src/dstore/pmix_esh.c

```diff
@@ -434,7 +723,7 @@ int _esh_fetch(const char *nspace, pmix_rank_t rank, const char *key, pmix_value
                     "%s:%d:%s:  no data for ns %s is found in the shared memory.",
                     __FILE__, __LINE__, __func__, nspace));
         /* unset lock */
-        flock(_lockfd, LOCK_UN);
+        flock(_esh_nspace_lockfd(nspace), LOCK_UN);
```

Comment:
This will help to avoid improper cleanups and restricts the number of lines at error detection to 2.